### ISSUE TITLE
Validation on LFW

### ISSUE THREAD

**Issue by geyongtao** [Joker316701882/Additive-Margin-Softmax#7]
Hi, @Joker316701882. I use the CASIA-Webface as training set and the LFW dataset as validation set. However, during the training process, the accuracy on LFW dataset is always 50% and the selected threshold is always 0. To find the problem, I read the source code and find the euclidean distance is used to calculate the distance between the two embedded features. As I know, AM-Softmax is a angular-based face recognition algorithm. My problem is:

1. Is that ok just using the euclidean-based distance to evaluate?
2. Using the euclidean-based algorithm to evaluate is logically feasible, so what's problem for the fixed accuracy? Is there anything I forget to configure before training?

**Comment by GangqiangZhao** [comment]
def evaluate_with_no_cv(emb_array, actual_issame):
    thresholds = np.arange(-4, 4, 0.01)
    embeddings1 = emb_array[0::2]
    embeddings2 = emb_array[1::2]

    nrof_thresholds = len(thresholds)
    accuracys = np.zeros((nrof_thresholds))
    
    diff = np.subtract(embeddings1, embeddings2)
    dist = -np.sum(np.square(diff),1)

    for threshold_idx, threshold in enumerate(thresholds):
        _, _, accuracys[threshold_idx] = facenet.calculate_accuracy(threshold, dist, actual_issame)

    best_acc = np.max(accuracys)
    best_thre = thresholds[np.argmax(accuracys)]
    return best_acc,best_thre

**Comment by geyongtao** [comment]
Well, I have done a litte change to the code so that it can calulate the cos similarity between embeddings when evaluating. 

My reference is  https://github.com/wy1iu/sphereface/tree/master/test/code/evaluation.m.

**Comment by weihehdc** [comment]
Hi, @Joker316701882. I use the CASIA-Webface as training set and the LFW dataset as validation set. However, during the training process, the accuracy on LFW dataset is always 100% and the selected threshold is ok. I read the source code print the 'acc' in facenet.calculate_accuracy find that  the Acc gradually increases from 0 to 1. In train.evaluate_with_no_cv, you do those op：
    best_acc = np.max(accuracys)
    best_thre = thresholds[np.argmax(accuracys)]
    return best_acc,best_thre
so the acc is always 100%.
my problem is how to fix it and  it is ok to get thres without cross-validation?
